src/client.ts

```diff
@@ -355,6 +364,74 @@ export class Client {
     }
   }
 
+  /**
+ * Take a screenshot of the page and save it to `filename` in `path` folder, with a `format` and `quality` (jpeg format only)
+ * If `selector` is passed in, it will take a screenshot of only that element
+ * and its children as opposed to the whole page.
+ *
+ * @param path - The path of where to save the screenshot to
+ * @param options - options
+ * @param options.filename - Name to be given to the screenshot. Optional
+ * @param options.selector - Screenshot the given selector instead of the full page. Optional
+ * @param options.format - The Screenshot format(and hence extension). Allowed values are "jpeg", "png", and "webp" - Optional
+ * @param options.quality - The image quality from 0 to 100, default 80. Applicable only if no format provided or format is "jpeg" - Optional
+ */
+  public async takeScreenshot(
+    path: string,
+    options?: {
+      selector?: string;
+      fileName?: string;
+      format?: "jpeg" | "png" | "webp";
+      quality?: number;
+    },
+  ): Promise<void> {
+    if (!existsSync(path)) {
+      await this.done();
+      throw new Error(`The provided folder path - ${path} doesn't exist`);
+    }
+    let ext = "jpeg";
+    let quality: number | undefined = 80;
+    // deno-lint-ignore ban-types
+    let clip: Object | undefined = undefined;
```

Comment:
can we do `const clip options!.selector ? await this.getViewport(options.selector) : undefined`?

Reply:
Surely

Reply:
Done